### PR TITLE
[Merged by Bors] - Add `ListDirectoryGroups` Operation

### PR DESCRIPTION
**Pull request by sheldonvaughn**

### CODE REVIEW

File: src/directory_sync/operations/list_directory_groups.rs [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+    /// The pagination options to use when listing directory users.
+    #[serde(flatten)]
+    pub pagination: PaginationOptions<'a>,
+
+    /// The filter to use when listing directory users.
+    #[serde(flatten)]
+    pub filter: DirectoryGroupsFilter<'a>,
+}
+
+/// [WorkOS Docs: List Directory Users](https://workos.com/docs/reference/directory-sync/group/list)
```
Comment (maxdeviant) [inline src/directory_sync/operations/list_directory_groups.rs]:
```suggestion
/// [WorkOS Docs: List Directory Groups](https://workos.com/docs/reference/directory-sync/group/list)
```

File: src/directory_sync/operations/list_directory_groups.rs [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+#[async_trait]
+pub trait ListDirectoryGroups {
+    /// Retrieves a list of [`DirectoryGroup`]s.
+    ///
+    /// [WorkOS Docs: List Directory Users](https://workos.com/docs/reference/directory-sync/group/list)
```
Comment (maxdeviant) [inline src/directory_sync/operations/list_directory_groups.rs]:
```suggestion
    /// [WorkOS Docs: List Directory Groups](https://workos.com/docs/reference/directory-sync/group/list)
```

File: src/directory_sync/operations/list_directory_groups.rs [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+    User {
+        /// The ID of the directory user to retrieve directory groups for.
+        user: &'a DirectoryUserId,
+    },
+}
+
+/// The options for [`ListDirectoryGroups`].
+#[derive(Debug, Serialize)]
+pub struct ListDirectoryGroupsOptions<'a> {
+    /// The pagination options to use when listing directory users.
```
Comment (maxdeviant) [inline src/directory_sync/operations/list_directory_groups.rs]:
```suggestion
    /// The pagination options to use when listing directory groups.
```

File: src/directory_sync/operations/list_directory_groups.rs [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+    #[serde(flatten)]
+    pub pagination: PaginationOptions<'a>,
+
+    /// The filter to use when listing directory users.
```
Comment (maxdeviant) [inline src/directory_sync/operations/list_directory_groups.rs]:
```suggestion
    /// The filter to use when listing directory groups.
```

**Review by maxdeviant — APPROVED** [review]
Just a couple tweaks to the doc comments 👍🏻

**Comment by sheldonvaughn** [comment]
bors merge

**Pull request closed without merging**